app/Filament/Resources/SystemUserResource/Pages/CreateSystemUser.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/*
+<COPYRIGHT>
+
+    Copyright © 2022-2023, Canyon GBS LLC. All rights reserved.
+
+    Advising App™ is licensed under the Elastic License 2.0. For more details,
+    see https://github.com/canyongbs/advisingapp/blob/main/LICENSE.
+
+    Notice:
+
+    - You may not provide the software to third parties as a hosted or managed
+      service, where the service provides users with access to any substantial set of
+      the features or functionality of the software.
+    - You may not move, change, disable, or circumvent the license key functionality
+      in the software, and you may not remove or obscure any functionality in the
+      software that is protected by the license key.
+    - You may not alter, remove, or obscure any licensing, copyright, or other notices
+      of the licensor in the software. Any use of the licensor’s trademarks is subject
+      to applicable law.
+    - Canyon GBS LLC respects the intellectual property rights of others and expects the
+      same in return. Canyon GBS™ and Advising App™ are registered trademarks of
+      Canyon GBS LLC, and we are committed to enforcing and protecting our trademarks
+      vigorously.
+    - The software solution, including services, infrastructure, and code, is offered as a
+      Software as a Service (SaaS) by Canyon GBS LLC.
+    - Use of this software implies agreement to the license terms and conditions as stated
+      in the Elastic License 2.0.
+
+    For more information or inquiries please visit our website at
+    https://www.canyongbs.com or contact us via email at legal@canyongbs.com.
+
+</COPYRIGHT>
+*/
+
+namespace App\Filament\Resources\SystemUserResource\Pages;
+
+use Filament\Forms\Form;
+use App\Models\SystemUser;
+use Filament\Forms\Components\TextInput;
+use Filament\Resources\Pages\CreateRecord;
+use App\Filament\Resources\SystemUserResource;
+
+class CreateSystemUser extends CreateRecord
+{
+    protected static string $resource = SystemUserResource::class;
+
+    public function form(Form $form): Form
+    {
+        return $form->schema([
+            TextInput::make('name')
+                ->required()
+                ->string(),
+        ]);
+    }
+
+    public function afterCreate(): void
+    {
+        /** @var SystemUser $systemUser */
+        $systemUser = $this->getRecord();
+
+        $token = str($systemUser->createToken('api')->plainTextToken)->after('|')->toString();
+
+        session()->put('apiToken', $token);
```

Comment:
Giving this some more thought I worry about the security implications of having the API token be stored in plain text somewhere. Even if it is the session for split seconds.

If there is no more secure way to do this within the request lifecycle of the view/edit page, then we may just want to have generating it be an Action or something like that on the view/edit page.